packages/userstorage/src/browser/user-storage-uri.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import URI from '@theia/core/lib/common/uri';
+
+export namespace UserStorageUri {
+
+    export const SCHEME = 'user_storage';
+
+    /**
+     * Creates a new file URI from the filesystem path argument.
+     * @param fsPath the filesystem path.
+     */
+    export function create(fsPath: string) {
+        const fsUri = new URI(fsPath);
+        return new URI('').withScheme(SCHEME).withPath(fsUri.path.base);
```

Comment:
fsPath is not a path but a 'uri as string'. 
The js doc is also wrong in that it doesn't create a file uri, but a user-storage uri.

Also I don't see how it translates e.g. a 
`file:///Users/username/.theia/settings.json` to
`user_storage:///settings.json`.
Seems like it would only replace the scheme, which is wrong, isn't it?

Shouldn't it be the symmetric function to below's `toFsUri`? If that is the case a better name e.g. `fromFsUri` would be good, too.

And then these conversion functions, belong into the `UserStorageFileSystemImpl` module.

Reply:
@sven it creates a new uri with an empty string i.e `new URI('')`, adds the user storage scheme and then the base of the file uri i.e `settings.json` in this case.

I can move those functions to the user storage impl, and in that case I guess they should become private members of the impl as they are not used anywhere else.

Reply:
Also in that case maybe fromFsUri is indeed a better name for the first method.